presentation/src/main/java/com/android254/presentation/common/components/SessionsCard.kt

```diff
@@ -13,4 +13,172 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.android254.presentation.common.components
\ No newline at end of file
+package com.android254.presentation.common.components
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.rounded.StarOutline
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.constraintlayout.compose.ConstraintLayout
+import coil.compose.AsyncImage
+import com.android254.presentation.models.SessionPresentationModel
+
+@Composable
+fun SessionsCard(sessionPresentationModel: SessionPresentationModel, onclick: () -> Unit) {
+    Card(
+        modifier = Modifier
+            .fillMaxWidth(0.95f)
+            .wrapContentHeight(),
+        shape = RoundedCornerShape(5),
+        elevation = CardDefaults.cardElevation(defaultElevation = 4.dp),
+        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
+        onClick = onclick
+    ) {
+        Row(
+            Modifier
+                .fillMaxWidth().wrapContentHeight()
+                .padding(PaddingValues(top = 20.dp))
+        ) {
+            SessionTimeComponent(sessionPresentationModel.sessionTime)
+            SessionDetails(sessionPresentationModel = sessionPresentationModel)
+        }
+    }
+}
+
+@Composable
+fun RowScope.SessionTimeComponent(time: String) {
+    val timeValue = mutableListOf<Char>()
+    val meridian = mutableListOf<Char>()
+    time.toList().reversed().forEach {
+        if (!it.isDigit() && meridian.size != 2) {
+            meridian.add(it)
+        } else {
+            timeValue.add(it)
+        }
```

Comment:
Let's have this calculated while mapping from domain to presentation so that we can test the logic. Ideally the UI should only set the string.

Reply:
You can have the `sessionStartTime`, `sessionEndTime` and `isStarred` in the presentation module.

Reply:
Sure.

Reply:
> Let's have this calculated while mapping from domain to presentation so that we can test the logic. Ideally the UI should only set the string.

For this, I was trying to separate the time and the am/pm bit e.g 10:15PM would then be 10:15  and the PM.